libpod/container_internal.go

```diff
@@ -1552,5 +1552,5 @@ func (c *Container) setupOCIHooks(ctx context.Context, g *generate.Generator) (e
 		return nil, nil
 	}
 
-	return manager.Hooks(g.Spec(), c.Spec().Annotations, len(c.config.UserVolumes) > 0)
+	return manager.Hooks(config, config.Annotations, len(c.config.UserVolumes) > 0)
```

Comment:
This is a semantic change from before - we were using the annotations from the config supplied when creating the container libpod, not the annotations in the final/completed spec used to initialize the OCI runtime. I don't necessarily think this is a bad change - it's counterintuitive for people to see an environment variable in the container at runtime but not be able to trigger hooks off it - but work noting.

Reply:
> This is a semantic change from before...

Oops, I only meant to adjust the first argument.  Fixed with 4f49ffd5 -> 4e7d15f8.

> I don't necessarily think this is a bad change...

I don't mind if you want to file a follow-up PR to make this semantic pivot; but I don't think it should be mixed up with this PR's refactor.